### PR TITLE
Segmented sieve - doctests

### PR DESCRIPTION
**Pull request by quant12345**
### Describe your change:

Added doctests.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: maths/segmented_sieve.py [hunk 1]
```diff
@@ -4,7 +4,22 @@
...
+    [2, 3, 5, 7]
+
+    >>> sieve(3**3)
+    [2, 3, 5, 7, 11, 13, 17, 19, 23]
+
+    >>> sieve()
+    Traceback (most recent call last):
+        ...
+    TypeError: sieve() missing 1 required positional argument: 'n'
+
```
Comment (tianyizheng02) [inline maths/segmented_sieve.py:21]:
```suggestion
```
I think it's fine to assume that the function will be called on the right number of inputs, so I don't think this test is necessary

File: maths/segmented_sieve.py [hunk 2]
```diff
@@ -4,7 +4,22 @@
 
 
 def sieve(n: int) -> list[int]:
-    """Segmented Sieve."""
+    """
+    Segmented Sieve.
+
+    Examples:
```
Comment (tianyizheng02) [inline maths/segmented_sieve.py:10]: Please add a doctest where the input itself is a prime number. For example,
```py
>>> sieve(29)
[2, 3, 5, 7, 11, 13, 17, 19, 23, 29]
```
This is to clarify whether the algorithm includes the input number itself or not.

Comment (tianyizheng02) [reply maths/segmented_sieve.py:10]: In addition, please make sure that the function can correctly handle negative and zero inputs

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

File: maths/segmented_sieve.py (hunk 2 above)
Comment (quant12345) [reply maths/segmented_sieve.py:10]: @tianyizheng02 corrected.

File: maths/segmented_sieve.py [hunk 3]
```diff
@@ -42,4 +61,9 @@ def sieve(n: int) -> list[int]:
     return prime
 
 
-print(sieve(10**6))
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
+
+    print(sieve(10**6))
```
Comment (cclauss) [inline maths/segmented_sieve.py:69]:
```suggestion
    print(f"{sieve(10**6) = }")
```

File: maths/segmented_sieve.py [hunk 4]
```diff
@@ -4,7 +4,26 @@
...
+    Examples:
+    >>> sieve(8)
+    [2, 3, 5, 7]
+
+    >>> sieve(27)
+    [2, 3, 5, 7, 11, 13, 17, 19, 23]
+
+    """
+
+    if n < 0:
```
Comment (cclauss) [inline maths/segmented_sieve.py:19]: I would be OK to combine these into a single `most be a positive integer`.

What happens on floating point input?

Comment (quant12345) [reply maths/segmented_sieve.py:19]: @cclauss 
`print(f"{sieve(87.5) = }")`
Output
```
temp = [True] * (high - low + 1)
TypeError: can't multiply sequence by non-int of type 'float'
```

Comment (cclauss) [reply maths/segmented_sieve.py:19]: Nice doctest to add.

Comment (quant12345) [reply maths/segmented_sieve.py]: @cclauss question is off topic. For some reason, my old commits end up in pull requests. I don’t understand why this is so? Every time I make a new branch and they still get there.

Comment (cclauss) [reply maths/segmented_sieve.py]: I will look in a few hours…

**Review by cclauss — APPROVED** [review]

**Review by cclauss — APPROVED** [review]
Nice!

**Review by dhruvmanila — APPROVED** [review]

**Review by tianyizheng02 — APPROVED** [review]

**Comment by quant12345** [comment]
@cclauss @tianyizheng02 @dhruvmanila thank you for your patience and advice.

**Pull request merged**